tests/match_source/build.bp

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019 Arm Limited.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+bob_binary {
+    name: "match_source_bin",
+    srcs: [
+        "main.c",
+        "data.txt",
+    ],
+    ldflags: ["-Wl,-dynamic-list,{{match_srcs \"*.txt\"}}"],
+}
+
+bob_generate_source {
+    name: "match_source_gen",
+    srcs: [
+        "main.c",
+        "custom.txt",
+    ],
+    out: ["custom_src_gen"],
+    cmd: "cat {{match_srcs \"custom.txt\"}} {{match_srcs \"main.c\"}} > $out",
```

Comment:
This would still pass if `{{match_srcs}}` returned e.g. an empty string, or e.g. matched the same source file twice.

Maybe we should write some C code in `custom.txt` too, then compile the generated code in another `bob_binary` which depends on the code in `main.c` as well as `custom.txt`.

Reply:
Yeah that's a good idea and I believe this should be made as a follow-up change